core/src/edu/cornell/gdiac/optimize/InputController.java

```diff
@@ -147,35 +147,64 @@ private void readGamepad() {
 	 *
 	 * @param secondary true if the keyboard should give priority to a gamepad
 	 */
+
+	//Arrays to registering switch and trigger presses
+			//We need to track their previous values so that we dont register a hold as repeated clicks
 	private boolean[] triggers;
 	private boolean[] triggerLast;
 	private boolean[] switches;
 	private boolean[] switchesLast;
 
 	boolean[] triggerPress;
 	boolean[] triggerLifted;
+	/**
+	 * Key for random indicator  on reset
+	 */
 	boolean rKey = false;
 	private void readKeyboard(boolean secondary) {
-		// Give priority to gamepad results
-		resetPressed = (secondary && resetPressed) || (Gdx.input.isKeyPressed(Input.Keys.T));
+		// Give priority to gamepad results, get input from keyboard
+		resetPressed = (secondary && resetPressed) || (Gdx.input.isKeyPressed(Input.Keys.M));
 		rKey = resetPressed && Gdx.input.isKeyPressed(Input.Keys.H);
 
-		triggerPress = new boolean[]{ Gdx.input.isKeyPressed(Input.Keys.D),
-												Gdx.input.isKeyPressed(Input.Keys.F),
-												Gdx.input.isKeyPressed(Input.Keys.J),
-												Gdx.input.isKeyPressed(Input.Keys.K)};
-
-		boolean[] switchesPress = new boolean[]{ Gdx.input.isKeyPressed(Input.Keys.E),
-												 Gdx.input.isKeyPressed(Input.Keys.R),
-												 Gdx.input.isKeyPressed(Input.Keys.U),
-												 Gdx.input.isKeyPressed(Input.Keys.I)};
-
-		for(int i = 0; i < 4; ++i){
-			triggers[i] = !triggerLast[i] && triggerPress[i];
-			switches[i] = !switchesLast[i] && switchesPress[i];
-			triggerLifted[i] = triggerLast[i] && !triggerPress[i];
-			triggerLast[i] = triggerPress[i];
-			switchesLast[i] = switchesPress[i];
+		triggerPress = new boolean[]{
+				Gdx.input.isKeyPressed(Input.Keys.Q),
+				Gdx.input.isKeyPressed(Input.Keys.W),
+				Gdx.input.isKeyPressed(Input.Keys.E),
+				Gdx.input.isKeyPressed(Input.Keys.R),
+				Gdx.input.isKeyPressed(Input.Keys.T),
+				Gdx.input.isKeyPressed(Input.Keys.Y),
+				Gdx.input.isKeyPressed(Input.Keys.U),
+				Gdx.input.isKeyPressed(Input.Keys.I),
+				Gdx.input.isKeyPressed(Input.Keys.O),
+				Gdx.input.isKeyPressed(Input.Keys.P)
+		};
+
+		boolean[] switchesPress = new boolean[]{
+				Gdx.input.isKeyPressed(Input.Keys.NUM_1),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_2),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_3),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_4),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_5),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_6),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_7),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_8),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_9),
+				Gdx.input.isKeyPressed(Input.Keys.NUM_0)
+		};
```

Comment:
I'm assuming you're doing this for the abstraction? I genuinely do not believe we need this many band members and lanes. I think the abstraction is fine, it's just that it doesn't make sense to have this many for the sake of being able to say we can do it.

Feel free to correct me if I'm wrong on this, both for this function and your rationale.